test/regress/regress0/smtlib/reset-assertions.smt2

```diff
@@ -0,0 +1,11 @@
+; EXPECT: unsat
+; EXPECT: sat
+(set-logic QF_BV)
+(declare-fun v0 () (_ BitVec 4))
+(set-option :produce-models true)
+(reset-assertions)
+(set-option :incremental true)
+(assert (and (= v0 (_ bv0 4)) (distinct v0 (_ bv0 4))))
+(check-sat)
+(reset-assertions)
+(check-sat)
```

Comment:
Can we have an additional unit test that tests `global-declarations` with `reset-assertion`? Something like:

```
(set-option :global-declarations true)
(set-logic QF_BV)
(declare-const x (_ BitVec 4))
(assert (= x (bvneg x)))
(check-sat)
(reset-assertions)
(assert (= x x))
(check-sat)
```

Reply:
I added a new test (a different one from above), and I also moved an existing test into `smtlib`. https://github.com/CVC4/CVC4/pull/3988/commits/871f43634aa3602e8e5d46da043a813168249f04